frontend/src/components/common/form/DateTimePicker.tsx

```diff
@@ -0,0 +1,70 @@
+import type { ReactElement } from "react";
+import React, { useRef, useState } from "react";
+import { HStack } from "@chakra-ui/react";
+
+import { combineDateAndTime } from "../../../utils/DateUtils";
+
+import DatePicker from "./DatePicker";
+import TimePicker from "./TimePicker";
+
+type DateTimePickerProps = {
+  name?: string;
+  onChange: (date: Date | null) => void;
+  value: Date | null | undefined;
+  isDisabled?: boolean;
+};
+
+const DateTimePicker = ({
+  isDisabled,
+  onChange,
+  value,
+  name,
+}: DateTimePickerProps): ReactElement => {
+  const timePickerRef = useRef<HTMLDivElement>(null);
+
+  const [internalDateValue, setInternalDateValue] = useState<Date | null>(null);
+  const [internalTimeValue, setInternalTimeValue] = useState<Date | null>(null);
+  const dateValue = value ?? internalDateValue;
+  const timeValue = value ?? internalTimeValue;
+
+  const handleDateChange = (newDate: Date | null) => {
+    setInternalDateValue(newDate);
+    if (timeValue && newDate) {
+      onChange(combineDateAndTime(newDate, timeValue));
+    }
+
+    // For some reason, the time picker doesn't focus properly if we click it
```

Comment:
what does this mean exactly?

Reply:
also, when the date picker is open and i click the hours and then the minutes, both stay open
<img width="299" alt="image" src="https://github.com/uwblueprint/jump-math/assets/69697180/11adc531-93e1-4428-a3d9-7be780179feb">

Reply:
Nice catch with the bug! It's a similar issue but not strictly related. Essentially, the problem is that the date picker component we chose to use steals focus when it starts to close, so we need to make sure we only request focus in other components AFTER it has already started closing. The comment above is to fix the problem when you TAB into the time picker; I added a similar fix for when you click it directly, which is the bug you found.